scipy/special/_logsumexp.py

```diff
@@ -114,8 +118,12 @@ def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
     with np.errstate(divide='ignore'):
         s = np.sum(tmp, axis=axis, keepdims=keepdims)
         if return_sign:
-            sgn = np.sign(s)
-            s *= sgn  # /= makes more sense but we need zero -> zero
+            # For complex, use the numpy>=2.0 convention for sign.
+            if np.issubdtype(s, np.complexfloating):
```

Comment:
I don't know what the scipy convention is here, but in astropy we'd explicitly do something like
```
if NUMPY_LT_2_0 and ...:
```
so that once the minimum supported version of numpy becomes 2.0, we have an easy way of
removing the work-around (since then `np.sign` itself will do the right thing in numpy >= 2.0).

Reply:
I couldn't find any examples of this in the library (aside from a couple tests and imports) so I decided not to introduce that kind of logic.